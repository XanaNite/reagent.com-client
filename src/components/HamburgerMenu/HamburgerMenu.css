ul li {
    list-style-type: none;
}

#page-wrap {
  text-align: center;

  /* Prevent sidebar from showing a scrollbar on page */
  overflow: auto;
}

.bm-menu-wrap{
    margin-left: -40px;
}

/* Individual item */
.bugerLink {  
  /* Our sidebar item styling */
  text-decoration: none;
  margin-bottom: 10px;
  color: black;
  transition: color 0.2s;
}

/* Change color on hover */
.bm-item:hover {
  color: white;
}

/* Position and sizing of burger button */
.bm-burger-button {
  position: fixed;
  width: 30px;
  height: 30px;
  left: 25px;
  top: 25px;
}

/* Color/shape of burger icon bars */
.bm-burger-bars {
  background: black;
}

/* Position and sizing of clickable cross button */
.bm-cross-button {
  margin: 10px;
  height: 24px;
  width: 24px;
}

/* Color/shape of close button cross */
.bm-cross {
  background: black;
}

/* General sidebar styles */
.bm-menu {
  background: whitesmoke;
  padding: 25px 0;
  font-size: 24px;
}

.bm-menu nav {
    height: auto !important;
}

/* Wrapper for item list */
.bm-item-list {
  color: black;
}

/* Styling of overlay */
.bm-overlay {
  background: whitesmoke;
}

@media screen and (min-width: 760px){
  .bm-menu-wrap{
    margin-left: -70px;
  }
}

@media screen and (min-width: 1050px){
  .bm-menu-wrap{
    margin-left: -110px;
  }
}