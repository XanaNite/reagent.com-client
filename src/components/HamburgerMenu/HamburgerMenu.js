import React from 'react'
import {Link} from 'react-router-dom'
import {slide as Menu} from 'react-burger-menu'
import './HamburgerMenu.css'

export default props => {
    return(
        <Menu>
            <ul>
                <li><Link to='/' className='bugerLink'>REagent Home</Link></li>
                <li><Link to='/dashboard' className='bugerLink'>Dashboard Home</Link></li>
                <li><Link to='/myprofile' className='bugerLink'>Update Your Profile</Link></li>
            </ul>
        </Menu>
    );
}