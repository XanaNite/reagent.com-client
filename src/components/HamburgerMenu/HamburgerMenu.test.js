import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HamburgerMenu from './HamburgerMenu';

describe(`HamburgerMenu component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders a form.HamburgerMenu by default', () => {
    const wrapper = shallow(<HamburgerMenu />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the HamburgerMenu given props', () => {
    const wrapper = shallow(<HamburgerMenu {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
