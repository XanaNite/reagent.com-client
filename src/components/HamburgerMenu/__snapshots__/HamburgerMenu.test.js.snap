// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenu component renders a form.HamburgerMenu by default 1`] = `
<Menu
  bodyClassName=""
  burgerBarClassName=""
  burgerButtonClassName=""
  className=""
  crossButtonClassName=""
  crossClassName=""
  disableAutoFocus={false}
  disableCloseOnEsc={false}
  htmlClassName=""
  id=""
  itemClassName=""
  itemListClassName=""
  menuClassName=""
  morphShapeClassName=""
  noOverlay={false}
  noTransition={false}
  onIconHoverChange={[Function]}
  onStateChange={[Function]}
  outerContainerId=""
  overlayClassName=""
  pageWrapId=""
  styles={Object {}}
  width={300}
>
  <ul>
    <li>
      <Link
        className="bugerLink"
        to="/"
      >
        REagent Home
      </Link>
    </li>
    <li>
      <Link
        className="bugerLink"
        to="/dashboard"
      >
        Dashboard Home
      </Link>
    </li>
    <li>
      <Link
        className="bugerLink"
        to="/myprofile"
      >
        Update Your Profile
      </Link>
    </li>
  </ul>
</Menu>
`;

exports[`HamburgerMenu component renders the HamburgerMenu given props 1`] = `
<Menu
  bodyClassName=""
  burgerBarClassName=""
  burgerButtonClassName=""
  className=""
  crossButtonClassName=""
  crossClassName=""
  disableAutoFocus={false}
  disableCloseOnEsc={false}
  htmlClassName=""
  id=""
  itemClassName=""
  itemListClassName=""
  menuClassName=""
  morphShapeClassName=""
  noOverlay={false}
  noTransition={false}
  onIconHoverChange={[Function]}
  onStateChange={[Function]}
  outerContainerId=""
  overlayClassName=""
  pageWrapId=""
  styles={Object {}}
  width={300}
>
  <ul>
    <li>
      <Link
        className="bugerLink"
        to="/"
      >
        REagent Home
      </Link>
    </li>
    <li>
      <Link
        className="bugerLink"
        to="/dashboard"
      >
        Dashboard Home
      </Link>
    </li>
    <li>
      <Link
        className="bugerLink"
        to="/myprofile"
      >
        Update Your Profile
      </Link>
    </li>
  </ul>
</Menu>
`;
